src/components/Text.tsx

```diff
@@ -76,30 +77,50 @@ Text.ALIGNS = ALIGNS;
 Text.allowedColors = allowedColors;
 Text.DEFAULT_PROPS = DEFAULT_PROPS;
 
-// @ts-ignore
-function Text(props) {
+type BaseTextProps = {
+  id?: string;
+  as?: OneOf<typeof AS>;
+  align?: OneOf<typeof ALIGNS>;
+  wrap?: boolean;
+  role?: string;
+  children: React.ReactNode;
+  testId?: string;
+} & ResponsiveProp<"margin", string>;
+
+type ConstraintTextProp<
+  T extends readonly any[],
+  U extends readonly any[]
```

Comment:
I can't. The point of this was not to have to be able to pass the typeof array in here. see below:
```ts
ConstraintTextProp<
  typeof allowedColors[0]["allowedColors"],
  typeof allowedColors[0]["textStyles"]
>
```

Instead of:
```ts
ConstraintTextProp<
  OneOf<typeof allowedColors[0]["allowedColors"]>,
  OneOf<typeof allowedColors[0]["textStyles"]>
>
```

Given that it is only used internally as a helper for the Text props I felt like having it be more concise was better.